### PR TITLE
chore(experimental): merge `v0.38.x` into `v0.38.x-experimental`

### PR DESCRIPTION
**Pull request by sergio-mena**
Getting `v0.38.x-experimental` up to date w.r.t. `v0.38.x`

THIS IS A MERGE COMMIT -- PLEASE DO NOT REVIEW FILE BY FILE (but rather the merge commit)

---

#### PR checklist

- ~[ ] Tests written/updated~
- ~[ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~
- ~[ ] Updated relevant documentation (`docs/` or `spec/`) and code comments~
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: store/bench_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/cometbft/cometbft/internal/test"
+	"github.com/cometbft/cometbft/types"
+	cmttime "github.com/cometbft/cometbft/types/time"
+)
+
+// TestLoadBlockExtendedCommit tests loading the extended commit for a previously
+// saved block. The load method should return nil when only a commit was saved and
+// return the extended commit otherwise.
+func BenchmarkRepeatedLoadSeenCommitSameBlock(b *testing.B) {
+	state, bs, cleanup := makeStateAndBlockStore()
```
Comment (cason) [inline store/bench_test.go:17]:
```suggestion
	state, bs, _, _, cleanup, _ := makeStateAndBlockStoreAndIndexers()
```

**Review by cason — COMMENTED** [review]
There is an error, suggested a solution.

File: store/bench_test.go (hunk 1 above)
Comment (sergio-mena) [reply store/bench_test.go:17]: Thanks! Will include this in the merge commit itself

File: state/store_test.go [hunk 2]
```diff
@@ -173,6 +173,7 @@ func TestPruneStates(t *testing.T) {
 						{Data: []byte{2}},
 						{Data: []byte{3}},
 					},
+					AppHash: make([]byte, 1),
```
Comment (cason) [inline state/store_test.go:176]: The same change should be added to `TestABCIResPruningStandalone` and `TestFinalizeBlockResponsePruning` as well.

**Review by cason — COMMENTED** [review]
Suggestion to how to fix other failed tests.

**Review by cason — APPROVED** [review]
Reviewed commit `1a0d4708831c1a4faadebdc5f93b7444a3e80b7b` and changes look harmless.  :)

File: state/store_test.go (hunk 2 above)
Comment (sergio-mena) [reply state/store_test.go:176]: Thanks, did it in 1a0d470883

**Pull request merged**